server/report/views.py

```diff
@@ -20,204 +22,173 @@ class ReportOwnerPermission(permissions.BasePermission):
+                question = TechnicalQuestion.objects.get(
+                    question_id=request.data["associated_id"]
                 )
-            
-            # check user is in the interview
-            interview = find_interview[0]
-            if not (interview.interviewer.id == self.request.data['reporter_user_id'] or interview.interviewee.id == self.request.data['reporter_user_id']):
+                report_data["associated_question"] = question
+
+            elif report_data["type"] == "member":
+                member = User.objects.get(id=request.data["associated_id"])
+                if member.id == reporter.id:
+                    return Response(
+                        {"error": "Cannot report yourself"},
+                        status=status.HTTP_400_BAD_REQUEST,
+                    )
+                report_data["associated_member"] = member
+
+            else:
                 return Response(
-                    {
-                        "error": "User not in interview"
-                    },
-                    status=status.HTTP_400_BAD_REQUEST
-                ) 
-        
-        serializer = ReportSerializer(data=self.request.data)
-        if serializer.is_valid():
-            serializer.save()
+                    {"error": "Invalid report type"}, status=status.HTTP_400_BAD_REQUEST
+                )
+            report = Report.objects.create(**report_data)
+            serializer = ReportSerializer(report)
+            return Response({"report": serializer.data}, status=status.HTTP_201_CREATED)
+
+        except User.DoesNotExist:
             return Response(
-                {
-                    "report": serializer.data
-                },
-                status=status.HTTP_201_CREATED
+                {"error": "Reporter user not found"}, status=status.HTTP_404_NOT_FOUND
             )
-        
-        return Response(
-            {
-                "error": serializer.errors
-            },
-            status=status.HTTP_400_BAD_REQUEST
-        )
+        except (Interview.DoesNotExist, TechnicalQuestion.DoesNotExist):
+            return Response(
+                {"error": f"{report_data['type'].capitalize()} not found"},
+                status=status.HTTP_404_NOT_FOUND,
+            )
+        except Exception as e:
+            return Response({"error": str(e)}, status=status.HTTP_400_BAD_REQUEST)
 
 
 class UpdateReportStatus(APIView):
     permission_classes = [IsAdmin]
+
     def patch(self, request, report_id):
-        
+
         report = Report.objects.filter(report_id=report_id)
         if not report:
             return Response(
-                {
-                    "error": "Report not found"
-                },
-                status=status.HTTP_404_NOT_FOUND
+                {"error": "Report not found"}, status=status.HTTP_404_NOT_FOUND
             )
 
-        if 'status' not in request.data:
+        if "status" not in request.data:
```

Comment:
Shouldn't we only allow certain statuses? Why can this be any value?

Reply:
yes, will add validation to serializer